star-dash/star-dash/Entities/EntityManager.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  EntityManager.swift
+//  star-dash
+//
+//  Created by Lau Rui han on 12/3/24.
+//
+
+import Foundation
+typealias ComponentSet = Set<ComponentId>
+class EntityManager {
+    var componentMap: [ComponentId: Component]
+    var entityMap: [EntityId: Entity]
+    var entityComponentMap: [EntityId: ComponentSet]
+    init(componentMap: [ComponentId : Component], entityMap: [EntityId : Entity], entityComponentMap: [EntityId: ComponentSet]) {
```

Comment:
I think can set up typealising for these complex types for improved readability

```
typealias ComponentMap = [ComponentId: Component]
typealias EntityMap = [EntityId: Entity]
typealias EntityComponentsMap = [EntityId: ComponentSet]
```

Reply:
changed